web/src/test/java/org/springframework/security/web/server/csrf/CsrfWebFilterTests.java

```diff
@@ -227,6 +229,21 @@ public void doFilterWhenTokenIsNullThenNoNullPointer() {
 				.isForbidden();
 	}
 
+	@Test
+	public void filterWhenSubscribingCsrfTokenMultipleTimesThenGenerateOnlyOnce() {
```

Comment:
I think this test passes whether or not the change in `CsrfWebFilter` is made. Will you please alter the test in such a way as to pass when your change to `CsrfWebFilter` is in place, but to fail when it isn't? I think `generateToken` might need to be instrumented differently.

Reply:
You are right. My confirmation was lacking.
Since `CsrfWebFilter` adds `Mono` generated from `generateToken` to the `ServerWebExchange` attribute, `Mono` itself is still cached.
I forgot that I have to pay attention to the number of times CSRF `Mono` is subscribed.

However, I could only think of `AtomicInteger` as an easy way to measure the number of calls to `generateToken`.
`MonoCacheTimeTest` was also testing the cache in the same way (https://github.com/reactor/reactor-core/blob/v3.4.6/reactor-core/src/test/java/reactor/core/publisher/MonoCacheTimeTest.java).
Another idea I came up with was to use `TestPublisher`, but I think this makes the test code harder to understand than it is now.